### PR TITLE
Converting the examples using javascript / node toolchain

### PR DESCRIPTION
**Pull request by obilodeau**
Just a start of javascript testing. People with javascript and npm experience please comment. I borrowed this from Asciidoctor.js and Asciidoctor-template.js.

Tests fail exactly where it should. Waiting on #143 to be fixed.

### CODE REVIEW

File: npm/builder.js [hunk 1]
```diff
@@ -67,3 +69,34 @@ Builder.prototype.removeDistDirSync = function () {
...
+
+Builder.prototype.examples = function (callback) {
+  const builder = this;
+
+  async.series([
+    callback => builder.build(callback), // Build
+    callback => builder.compileExamples(callback), // Compile examples
+  ], () => {
+    log.info(`
+In order to visualize the result, a local HTTP server must be started within the root of this project otherwise you will have cross-origin issues.
```
Comment (Mogztter) [inline npm/builder.js]: I don't think it applies in this context

File: npm/builder.js [hunk 2]
```diff
@@ -67,3 +69,34 @@ Builder.prototype.removeDistDirSync = function () {
...
+For this purpose, you can run the following command to start a HTTP server locally: 'npm run server'.`);
+    log.success(`You can now open: build/examples/`);
+    typeof callback === 'function' && callback();
+  });
+};
+
+Builder.prototype.compileExamples = function (callback) {
```
Comment (Mogztter) [inline npm/builder.js]: I think the naming is misleading since we are not really compiling examples

File: npm/examples.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const log = require('bestikk-log');
+const Builder = require('./builder.js');
+const builder = new Builder();
+
+log.task('examples');
+builder.examples();
```
Comment (Mogztter) [inline npm/examples.js]: As mentioned in https://github.com/asciidoctor/asciidoctor-reveal.js/pull/156#issuecomment-335613495 I think we should use an "examples" task to demonstrate how to use the converter in JavaScript.

In the Asciidoctor.js repository, I have one example with a textarea on the left of the page and the rendered Reveal.js presentation on the right of the page.

File: npm/builder.js [hunk 4]
```diff
@@ -48,7 +50,7 @@ Builder.prototype.removeBuildDirSync = function () {
 Builder.prototype.compile = function (callback) {
   log.task('compile');
   var opalCompiler = new OpalCompiler({dynamicRequireLevel: 'ignore'});
-  opalCompiler.compile('asciidoctor-revealjs', 'build/asciidoctor-revealjs.js', ['lib']);
+  opalCompiler.compile('asciidoctor-revealjs', 'build/asciidoctor-reveal.js', ['lib']);
```
Comment (Mogztter) [inline npm/builder.js]: You are probably right the `revealjs.js` is a bit too much :wink:

File: npm/builder.js (hunk 1 above)
Comment (obilodeau) [reply npm/builder.js]: Removed

File: npm/examples.js (hunk 3 above)
Comment (obilodeau) [reply npm/examples.js]: Not sure I understand what you mean. We have it in the documentation already. Adding an example  would mean maintaining two copies of the same code and instructions.

File: npm/builder.js (hunk 2 above)
Comment (obilodeau) [reply npm/builder.js]: Fixed. Changed to convert.

File: npm/examples.js (hunk 3 above)
Comment (Mogztter) [reply npm/examples.js]: Alright, I thought the purpose of the examples task was to provide a "live" example.

**Comment by Mogztter** [comment]
🎉 🎉 🎉

**Pull request merged**